src/encoded/static/components/static-pages/components/StatisticsPageViewBody.js

```diff
@@ -0,0 +1,1092 @@
+                const params = navigate.getBrowseBaseParams(props.browseBaseState || null);
+                if (props.currentGroupBy){
+                    params.group_by = props.currentGroupBy;
+                }
+                //if (props.browseBaseState === 'all') params['group_by'] = ['award.project'];
+                const uri = '/date_histogram_aggregations/?' + stringify(params) + '&limit=0&format=json';
+
+                // For local dev/debugging; don't forget to comment out if using.
+                //uri = 'https://data.4dnucleome.org' + uri;
+                return uri;
+            }
+        },
+        'shouldRefetchAggs' : function(pastProps, nextProps){
+            return StatsViewController.defaultProps.shouldRefetchAggs(pastProps, nextProps) || (
+                pastProps.browseBaseState !== nextProps.browseBaseState ||
+                pastProps.currentGroupBy  !== nextProps.currentGroupBy
+            );
+        }
+    };
+
+    constructor(props){
+        super(props);
+        this.fetchAndGenerateExternalTermMap = this.fetchAndGenerateExternalTermMap.bind(this);
+        this.state = { "externalTermMap" : null };
+    }
+
+    componentDidMount(){
+        this.fetchAndGenerateExternalTermMap();
+    }
+
+    componentDidUpdate(pastProps){
+        const { shouldRefetchAggs, session } = this.props;
+        if (shouldRefetchAggs(pastProps, this.props)){
+            if (pastProps.session !== session){
+                // Avoid triggering extra re-aggregation from new/unnecessary term map being loaded.
+                this.fetchAndGenerateExternalTermMap(true);
+            }
+        }
+    }
+
+    fetchAndGenerateExternalTermMap(refresh = false){
+        const { externalTermMap } = this.state;
+        if (!refresh && externalTermMap && _.keys(externalTermMap).length > 0) return;
+
+        ajax.load('/search/?type=Award&limit=all', (resp)=>{
+            this.setState({
+                'externalTermMap' : _.object(_.map(resp['@graph'] || [], function(award){
+                    return [ award.center_title, award.project !== '4DN' ];
+                }))
+            });
+        });
+    }
+
+    render(){ return <StatsViewController {...this.props} {...this.state} />; }
+
+}
+
+
+
+class UsageChartsCountByDropdown extends React.PureComponent {
+
+    constructor(props){
+        super(props);
+        this.handleSelection = this.handleSelection.bind(this);
+    }
+
+    handleSelection(evtKey, evt){
+        const { changeCountByForChart, chartID } = this.props;
+        changeCountByForChart(chartID, evtKey);
+    }
+
+    render(){
+        const { countBy, chartID } = this.props;
+        const currCountBy = countBy[chartID];
+
+        const menuOptions = new Map();
+
+        if (chartID === 'experiment_set_views' || chartID === 'file_views'){
+            menuOptions.set('views',        <React.Fragment><i className="icon fas icon-fw icon-eye mr-1"/>Detail View</React.Fragment>);
```

Comment:
In effect React.Fragment (and any other instance of JSX element/component usage) is itself a deferred function that, when finally ran by React, tells React to render a fragment of elements in the DOM... just in a somewhat unique syntax ('JSX'). And IMO existing JSX syntax already fits on 1 line + is explicit enough (w. no need to go to different place in code to lookup `generateReactFragment` to see what it does, though I guess the func would be located in same scope here). (This is probably subjective on my part idk -) Part of me might argue adding more lines of code in form of the `generateReactFragment` func isn't a perfect great trade-off to ctrl+f&replace in this case because is still relatively manageable number (i.e. <20, not 100+) (i.e. wherein had replaced "mr-05" with "mr-1" in this PR).

Reply:
I think there are some competing paradigms here that I think make both of us right - and obviously it's your discretion 😄. What you said makes a lot of sense though. To me (because I don't write JS all day) JSX really stands out and feels like something that should be encapsulated when really that's probably not practical/necessary in most cases.